src/helpers.c

```diff
@@ -130,60 +130,66 @@ void unlink_all_staged_content(struct file *file)
 	}
 }
 
+/* Ensure that a directory either doesn't exist
+ * or does exist and belongs to root.
+ * If it exists, but is not a directory or belonging to root, remove it
+ * returns true if it doesn't exist
+ */
+static int ensure_root_owned_dir(const char *dirname)
+{
+	struct stat sb;
+	int ret = stat(dirname, &sb);
+	if (ret && (errno == ENOENT)) {
+		return true;
+	}
+	if ((sb.st_uid == 0) &&
+	    (S_ISDIR(sb.st_mode)) &&
+	    ((sb.st_mode & 0777) == 0700)) {
+		return false;
+	}
+	/* Oops, not owned by root or
+	 * not a directory or wrong perms
+	 */
+	ret=swupd_rm(state_dir);
```

Comment:
I know users shouldn't be putting things in /var/lib/swupd, but should we at least print a warning that we are removing content before doing so? Or no since that should be swupd-managed anyways?